Automattic/wp-calypso
# Stats: remove the SITE_STATS_REQUEST_SUCCESS action

The action is always dispatched together with `SITE_STATS_RECEIVE`, so we can unify them into one. Dispatching two actions on every stats request success leads to wasted rerenders of `StatsModule` components. When loading a `/stats/day/:site` page, there are 11 stats request being issued and each `StatsModule` rerenders 4 times. This patch reduces that to just 3 rerenders. The optimizations will continue with further incremental patches 📈 

**How to test:**

Display stats for your favorite WordPress.com site:

<img width="717" alt="stats-refresh" src="https://user-images.githubusercontent.com/664258/35803280-6ff32682-0a73-11e8-8573-68401c5be15a.png">

- check that the UI shows a placeholder while loading and the actual data when finished loading.
- check that switching with the left and right arrows between days (or weeks or months) also works.
- check that when showing stats for the "current period" (today, this week, this month, ...) there is a "Last update" label and that it refreshes every 3 minutes.
